pkgs/development/compilers/llvm/common/default.nix

```diff
@@ -327,7 +327,11 @@ let
           )
           ++ lib.optional (
             lib.versionAtLeast metadata.release_version "16" && lib.versionOlder metadata.release_version "18"
-          ) (metadata.getVersionFile "lld/add-table-base.patch");
+          ) (metadata.getVersionFile "lld/add-table-base.patch")
+          # backports from LLVM tip-of-tree to latest stable for emscripten
+          ++ lib.optional (
+            lib.versionAtLeast metadata.release_version "18"
```

Comment:
Probably would be good to copy the patch to the LLVM git version as well or change this version check to only add the patch for LLVM 18. Otherwise, we'll get an eval error.

Reply:
Good catch -- apologies for missing that that'd screw it up. Adjusted with latest force-push.

Reply:
Sweet, that looks great.